apps/api/src/app/app.controller.ts

```diff
@@ -110,18 +112,19 @@ export class AppController {
   @ApiOperation({ summary: 'Create New Links' })
   @ApiBody({ type: Link })
   @ApiResponse({ type: Link, status: 200})
-  async register(@Body() link: Link): Promise<LinkModel> {
-    return this.appService.createLink(link);
+  async register(@Body() link: Link): Promise<LinkModel> { 
+      const response:Promise<link> =  this.appService.createLinkInDB(link);
+      return response;
   }
 
-  
+  // no way to change the params ??
   @Patch('update/:id')
   @ApiOperation({ summary: 'Update Existing Links' })
   @ApiBody({ type: Link })
   @ApiResponse({ type: Link, status: 200})
   async update(@Param('id') id: string, @Body() link: Link ): Promise<LinkModel> {
     return this.appService.updateLink({
-      where: { customHashId: id },
+      where: {customHashId: id},
       data: { 
         userID: link.user || null,
         tags: link.tags || null,
```

Comment:
update method implementation is not correct I think. it should be calling app service update method.

Reply:
@yuvrajsab 
Do you mean prisma update method as I cannot see any other method for update,
Also I have to modify the update link method to handle the case for I'd being custom hash I'd.
Also few edge cases where the user only want to change the link but not hashid is handled in it.